addons/hr_expense/models/account_move.py

```diff
@@ -25,3 +25,7 @@ def _payment_state_matters(self):
         if self.line_ids.expense_id:
             return True
         return super()._payment_state_matters()
+
+    def button_cancel(self):
+        super(AccountMove, self).button_cancel()
+        self.expense_sheet_id.expense_line_ids.refuse_expense('Cancelled by the account move related to this expense.')
```

Comment:
Last nitpick: we should translate the string (and add the new string into the .pot file as well) and maybe leave a more generic error as well
```
- self.expense_sheet_id.expense_line_ids.refuse_expense('Cancelled by the account move related to this expense.')
+ self.expense_sheet_id.expense_line_ids.refuse_expense(_('Payment Refused'))
```